contracts/mesh-provider/src/contract.rs

```diff
@@ -253,6 +259,39 @@ pub fn execute_unbond(
     Ok(Response::new().add_message(msg))
 }
 
+pub fn execute_claim_rewards(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+) -> Result<Response, ContractError> {
+    let sender = deps.api.addr_validate(info.sender.as_str())?;
+    let amount = REWARDS
+        .prefix(&sender)
+        .range(deps.storage, None, None, Order::Ascending)
+        .map(|res| -> Result<Coin, ContractError> {
+            let (denom, amount) = res?;
+
+            Ok(Coin { denom, amount })
+        })
+        .collect::<Result<Vec<Coin>, ContractError>>()?;
+
+    if amount.is_empty() {
+        return Err(ContractError::NoRewardsToClaim {});
+    }
+
+    let balance = deps.querier.query_all_balances(env.contract.address)?;
+
+    // let msg = BankMsg::Send {
+    //     to_address: sender.to_string(),
+    //     amount: amount.clone(),
+    // };
+
+    amount.iter().for_each(|coin| {
+        REWARDS.remove(deps.storage, (&sender, coin.denom.clone()));
+    });
+
+    Ok(Response::new().add_attribute("balances", balance.len().to_string()))
```

Comment:
Still need to include the `BankMsg`

Reply:
If i keep it it fails, theres no balance in provider.

Reply:
Might be able to finish it now that withdrawing to the provider is working better.